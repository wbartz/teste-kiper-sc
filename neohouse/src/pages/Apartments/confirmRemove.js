import PropTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router';
import Modal from '../../components/Modal';

export const ConfirmRemove = ({ show, onClose, onConfirm }) => {
  return (
    <Modal className="edit-modal" show={show} title="Remover?">
      <div className="row">
        <span>Tem certeza que deseja remover esse apartamento?</span>
        <p>Todos os seus moradores também serão removidos.</p>
      </div>
      <div className="row buttons confirm-buttons">
        <button type="button" className="btn btn-flat" onClick={onClose}>
          Cancelar
        </button>
        <button type="button" className="btn primary" onClick={onConfirm}>
          Remover
        </button>
      </div>
    </Modal>
  );
};

ConfirmRemove.propTypes = {
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

export default withRouter(ConfirmRemove);
