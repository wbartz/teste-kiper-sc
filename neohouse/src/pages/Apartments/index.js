import React, { useContext, useEffect, useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import { AppContext } from '../../containers/StoreProvider';
import Table from '../../components/Table';
import './index.scss';

const Apartments = ({
  removeApartment,
  getApartments,
  location,
  apartments,
  history,
}) => {
  const [hasData, setData] = useState(false);
  const getData = useCallback(async () => {
    const { block } = location.state;
    await getApartments(block);
    setData(true);
  }, [getApartments, location.state]);

  useEffect(() => {
    if (!hasData) {
      getData();
    }
  }, [location, getData, hasData]);

  const handleView = (id) =>
    history.push(`${location.pathname}/moradores/apartamento-${id}`, {
      block: location.state.block,
      id,
    });
  const handleEdit = (id) =>
    history.push(`${location.pathname}/edit/apartamento-${id}`, {
      block: location.state.block,
      id,
    });
  const handleDelete = (id) => removeApartment(id, getData);

  return (
    <div className="page">
      <div className="apartment-page">
        <Table
          header={['NÃºmero Apt.', 'ResponsÃ¡vel', 'E-mail', 'Telefone']}
          lines={apartments}
          onEdit={handleEdit}
          onView={handleView}
          onRemove={handleDelete}
        />
      </div>
    </div>
  );
};

Apartments.propTypes = {
  getApartments: PropTypes.func.isRequired,
  removeApartment: PropTypes.func.isRequired,
  apartments: PropTypes.oneOfType([PropTypes.array]),
  location: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
};

Apartments.defaultProps = {
  apartments: [],
};

export default (props) => <Apartments {...useContext(AppContext)} {...props} />;
