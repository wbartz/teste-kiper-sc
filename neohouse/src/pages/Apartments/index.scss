.confirm-buttons {
  display: flex;
  justify-content: space-between;
  margin-left: 100px;

  .primary {
    align-self: flex-end;
  }
}

.apartment {
  .search-field {
    .row {
      .col {
        padding: 0;

        .btn {
          margin-top: 25px;
        }
      }
    }
  }
}
