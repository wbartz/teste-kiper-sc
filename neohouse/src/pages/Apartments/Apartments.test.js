import { shallow } from 'enzyme';
import React from 'react';
import Apartments from '.';

describe('Apartments', () => {
  const props = {
    getApartments: jest.fn(),
    removeApartment: jest.fn(),
    location: {
      state: {
        block: 1,
      },
    },
    history: {
      push: jest.fn(),
    },
  };

  it('render without crashing', () => {
    expect(shallow(<Apartments {...props} />)).toMatchSnapshot();
  });
});
