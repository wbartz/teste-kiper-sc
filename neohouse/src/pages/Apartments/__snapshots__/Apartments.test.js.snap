// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apartments render without crashing 1`] = `ShallowWrapper {}`;
